cirq-google/cirq_google/engine/engine_processor_test.py

```diff
@@ -798,22 +799,13 @@ def test_list_reservations_time_filter_behavior(list_reservations):
 
 @mock.patch('cirq_google.engine.engine_client.EngineClient', autospec=True)
 def test_run_sweep_params(client):
```

Comment:
I think having to change these tests because of a feature flag is a bit of a code-smell, especially since they would have to be changed back if we flip the default value of the flag. You might consider instead putting the stream configuration on `EngineContext`, which has other things like the protocol version and serializer that can be changed. Then for tests we can exercise both the streaming and non-streaming execution by creating an appropriate context, and those tests will continue to work even if we change the default value of the flag.

Reply:
Thanks for the catch! Yeah I agree this is a code smell. Using `EngineContext` sounds good to me as long as it's not user-facing. Is this true for this class?

Reply:
EngineContext is not typically used directly by end-users. It's used to share API state between the various resource classes.